cudaextender/include/claraparabricks/genomeworks/cudaextender/extender.hpp

```diff
@@ -0,0 +1,151 @@
+    /// \param[in] target_length Length of target sequence
+    /// \param[in] score_threshold Score threshold for filtering extended segments
+    /// \param[in] d_seed_pairs Device pointer to array of seed pairs mapping between
+    ///                         target and query sequences
+    /// \param[in] num_seed_pairs Length of d_seed_pairs array
+    /// \param[out] d_scored_segment_pairs Pointer to a pre-allocated device location for
+    ///                                    storing extension output
+    /// \param[out] d_num_scored_segment_pairs Pointer to pre-allocated device location for
+    ///                                        storing length of extension output
+    /// \return Status of the async extension launch
+    virtual StatusType extend_async(const char* d_query, int32_t query_length,
+                                    const char* d_target, int32_t target_length,
+                                    int32_t score_threshold, SeedPair* d_seed_pairs,
+                                    int32_t num_seed_pairs, ScoredSegmentPair* d_scored_segment_pairs,
+                                    int32_t* d_num_scored_segment_pairs) = 0;
+
+    /// \brief Waits for CUDA accelerated extension to finish
+    ///
+    /// To be used with the host pointer extend_async API.
+    /// Blocking call that waits for all the extensions scheduled
+    /// on the GPU to come to completion.
+    ///
+    /// \return Synchronization and memory copy status
+    virtual StatusType sync() = 0;
+
+    /// \brief Returns the computed scored segment pairs
+    ///
+    /// To be used with the host pointer extend_async API. sync() must
+    /// be called after launching async extension and before calling this function
+    /// or the returned results will not be valid.
+    ///
+    /// \return Vector of Scored Segment Pairs that have a score >= score_threshold
+    virtual const std::vector<ScoredSegmentPair>& get_scored_segment_pairs() const = 0;
+
+    /// \brief Reset Extender object and free device/host memory
+    virtual void reset() = 0;
+};
+
+/// \brief Create Extender object
+///
+/// \param h_score_mat Host pointer to scoring matrix for use during extension
+/// \param score_mat_dim Dimension of the scoring matrix
+/// \param xdrop_threshold Threshold for performing X-Drop
+/// \param no_entropy Flag indicating whether to use entropy during extension
+/// \param stream CUDA Stream to be used with extension
+/// \param device_id GPU to be used for extension
+/// \param allocator DeviceAllocator to be used for allocating/freeing memory
+/// \param type Type of extension to be performed
+/// \return Unique pointer to Extender object.
+std::unique_ptr<Extender> create_extender(const int32_t* h_score_mat, const int32_t score_mat_dim, const int32_t xdrop_threshold, const bool no_entropy, cudaStream_t stream, const int32_t device_id, DefaultDeviceAllocator allocator, const ExtensionType type = ExtensionType::ungapped_xdrop);
```

Comment:
can this be made into a multi line definition?

Reply:
Done.